restapi/src/main/java/io/stargate/web/docsapi/service/DocsSchemaChecker.java

```diff
@@ -0,0 +1,48 @@
+package io.stargate.web.docsapi.service;
+
+import io.stargate.db.schema.Schema;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.exception.ErrorCode;
+import io.stargate.web.docsapi.exception.ErrorCodeRuntimeException;
+import io.stargate.web.docsapi.service.util.ImmutableKeyspaceAndTable;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A helper class for checking that particular keyspace/table combinations exist and are valid
+ * document API collections.
+ */
+public class DocsSchemaChecker {
+  private Schema lastCheckedSchema;
+
+  private final ConcurrentHashMap<ImmutableKeyspaceAndTable, Boolean> validatedDocCollections =
+      new ConcurrentHashMap<>();
+
+  private void clearCacheOnSchemaChange(DocumentDB db) {
+    if (!db.schema().equals(lastCheckedSchema)) {
```

Comment:
If we're comparing the entire schema on every call is it worth having a cache?

Reply:
According to Dmitri, Schema pre-hashes its equals / hashCode and so this comparison should be quick

Reply:
The idea is to short-circuit on stable schema. I'm not actually sure if it's going to have a measurable impact. So I'd be ok with removing the schema check. However, without it we'd have to refresh by time, which is prone to false positives and we really wanted to avoid false positive 500 errors.

We discussed using a schema change listener for invalidation, but it seems to be about as effective as a schema-global check in serverless env. We you have any suggestions for cache invalidation, they'd be very welcome, considering the proverbial complexity of the problem here :)

Reply:
After running the same perf tests using YCSB, I didn't find any major change in latency, so I'm going to merge this one as-is; if we think of a better way to do this caching in the future we can always change it :) no one-way doors here